pkg/utils/kubernetes/reduce/filters.go

```diff
@@ -315,21 +371,52 @@ func FilterHPAs(hpas []autoscalingv2.HorizontalPodAutoscaler) []autoscalingv2.Ho
 // Filter functions - ValidatingWebhookConfigurations
 // -----------------------------------------------------------------------------
 
-// filterValidatingWebhookConfigurations filters out the ValidatingWebhookConfigurations to be kept and returns
-// all the ValidatingWebhookConfigurations to be deleted. The oldest ValidatingWebhookConfiguration is kept.
-func filterValidatingWebhookConfigurations(webhookConfigurations []admregv1.ValidatingWebhookConfiguration) []admregv1.ValidatingWebhookConfiguration {
-	if len(webhookConfigurations) < 2 {
+// filterValidatingWebhookConfigurations filters out the ValidatingWebhookConfigurations
+// to be kept and returns all the ValidatingWebhookConfigurations to be deleted.
+// The following criteria are used:
+//  1. creationTimestamp (newer is better, because newer ValidatingWebhookConfiguration can contain new rules)
+//  2. using legacy labels (if present): if a ValidatingWebhookConfiguration does
+//     not have the legacy labels, it is considered newer and will be kept.
+func filterValidatingWebhookConfigurations(vwcs []admregv1.ValidatingWebhookConfiguration) []admregv1.ValidatingWebhookConfiguration {
+	if len(vwcs) < 2 {
 		return []admregv1.ValidatingWebhookConfiguration{}
 	}
 
-	best := 0
-	for i, webhookConfig := range webhookConfigurations {
-		if webhookConfig.CreationTimestamp.Before(&webhookConfigurations[best].CreationTimestamp) {
-			best = i
+	newestWithManagedByLabels := -1
+	newestLegacy := -1
+	for i, vwc := range vwcs {
+		labels := vwc.GetLabels()
+
+		_, okManagedBy := labels[consts.GatewayOperatorManagedByLabel]
+		_, okManagedByNs := labels[consts.GatewayOperatorManagedByNamespaceLabel]
+		_, okManagedByName := labels[consts.GatewayOperatorManagedByNameLabel]
+		if okManagedBy && okManagedByNs && okManagedByName {
+			if newestWithManagedByLabels == -1 {
+				newestWithManagedByLabels = i
+				continue
+			}
+
+			if vwc.CreationTimestamp.After(vwcs[newestWithManagedByLabels].CreationTimestamp.Time) {
+				newestWithManagedByLabels = i
+			}
+			continue
 		}
+
+		if newestLegacy == -1 {
+			newestLegacy = i
+			continue
+		}
+
+		if vwc.CreationTimestamp.After(vwcs[newestLegacy].CreationTimestamp.Time) {
+			newestLegacy = i
+		}
+		continue
 	}
 
-	return append(webhookConfigurations[:best], webhookConfigurations[best+1:]...)
+	if newestWithManagedByLabels != -1 {
+		return append(vwcs[:newestWithManagedByLabels], vwcs[newestWithManagedByLabels+1:]...)
+	}
+	return append(vwcs[:newestLegacy], vwcs[newestLegacy+1:]...)
```

Comment:
Here we're exposed to issue described in 
- https://github.com/Kong/gateway-operator/issues/146